pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/util.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cnsfileaccessconfig
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+
+	vmoperatortypes "github.com/vmware-tanzu/vm-operator-api/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	cnsfileaccessconfigv1alpha1 "sigs.k8s.io/vsphere-csi-driver/pkg/apis/cnsoperator/cnsfileaccessconfig/v1alpha1"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+	k8s "sigs.k8s.io/vsphere-csi-driver/pkg/kubernetes"
+)
+
+// getVirtualMachine gets the virtual machine instance with a name on a SV namespace
+func getVirtualMachine(ctx context.Context, vmName string, namespace string) (*vmoperatortypes.VirtualMachine, error) {
+	log := logger.GetLogger(ctx)
+	restClientConfig, err := k8s.GetKubeConfig(ctx)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to initialize rest clientconfig. Error: %+v", err)
+		log.Error(msg)
+		return nil, err
+	}
+
+	vmOperatorClient, err := k8s.NewClientForGroup(ctx, restClientConfig, vmoperatortypes.GroupName)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to initialize vmOperatorClient. Error: %+v", err)
+		log.Error(msg)
+		return nil, err
+	}
+
+	virtualMachine := &vmoperatortypes.VirtualMachine{}
+	vmKey := types.NamespacedName{
+		Namespace: namespace,
+		Name:      vmName,
+	}
+	if err := vmOperatorClient.Get(ctx, vmKey, virtualMachine); err != nil {
+		msg := fmt.Sprintf("Failed to get virtualmachine instance for the VM with name: %q. Error: %+v", vmName, err)
+		log.Error(msg)
+		return nil, err
+	}
+	return virtualMachine, nil
+}
+
+// setInstanceOwnerRef sets ownerRef on CnsFileAccessConfig instance to VM instance
+func setInstanceOwnerRef(instance *cnsfileaccessconfigv1alpha1.CnsFileAccessConfig, vmName string,
```

Comment:
Can this function be moved to `cnsoperator/util/util.go`? I see that other controller use/can use this too.

Reply:
Created this issue: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/587 for tracking the suggestion.
I am not addressing the review comment in this PR to avoid more complexity/testing cycles. It would need some additional effort to refactor 4 different Custom resource controller.go files and also run some tests.
Thanks